File: js/modules/student.js
export class Student  {
	constructor(name, grades, status) {
		this.name = name;
		this.grades= grades;
		this.status= status;
	}

	printStudent() {
		let cc = "[ ";
		for(let i = 0; i < this.grades.length; i++) {
			let grade = this.grades[i];
			cc += grade;
			if(i !== this.grades.length - 1) {
				cc += " - ";
			}
		} 
		console.log(`${this.name} ${cc}] ${this.status}`);
	}

	studentToString() {
		let cc = "[ ";
		for(let i = 0; i < this.grades.length; i++) {
			let grade = this.grades[i];
			cc += grade;
			if(i !== this.grades.length - 1) {
				cc += " - ";
			}
		} 
		return`${this.name} ${cc}] ${this.status}`;
	}

}

File: src/student.cpp
#include "student.h"
#include <ctime>
#include <ios>
#include <string>
#include <sstream>
#include <iomanip>

Student::Student(std::string name, std::string registrationCode, std::vector<float> grades, bool active):
	name{name}, registrationCode{registrationCode}, grades{grades}, active{active} {};

std::string Student::getName() const {
	return name;
};

std::string Student::getRegistration() const {
	return registrationCode; 
};

std::vector<float> Student::getGrades() const {
	return grades;
};

bool Student::isActive() const {
	return active;
};


std::ostream& operator<< (std::ostream& os, const Student& s) {
	std::string acc{};
	auto name{s.getName()};
	auto registration{s.getRegistration()};
	auto grades{s.getGrades()};
	auto active{Utils::boolToInt(s.isActive())};

	
	acc += ("student: " + name);
	acc += "\n";
	acc += ("registration: " + registration);
	acc += "\n";
	acc += ("active: " + active);
	acc += "\n";
	acc += "grades: [  ";
	for(auto grade : grades) {
		std::stringstream stream;
		stream << std::fixed << std::setprecision(2) << grade;
		auto current = stream.str();
		acc += current;
		acc += " ";
	}
	acc += " ]";

	os << acc;
	return os;
};
